tfhe/src/integer/server_key/radix_parallel/scalar_mul.rs

```diff
@@ -313,4 +415,104 @@ impl ServerKey {
     ) {
         *ctxt = self.smart_scalar_mul_parallelized(ctxt, scalar);
     }
+
+    /// Computes homomorphically a multiplication between a scalar and a ciphertext.
+    ///
+    /// This function, like all "default" operations (i.e. not smart, checked or unchecked), will
+    /// check that the input ciphertexts block carries are empty and clears them if it's not the
+    /// case and the operation requires it. It outputs a ciphertext whose block carries are always
+    /// empty.
+    ///
+    /// This means that when using only "default" operations, a given operation (like add for
+    /// example) has always the same performance characteristics from one call to another and
+    /// guarantees correctness by pre-emptively clearing carries of output ciphertexts.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2;
+    ///
+    /// // We have 4 * 2 = 8 bits of message
+    /// let modulus = 1 << 8;
+    /// let size = 4;
+    /// let (cks, sks) = gen_keys_radix(&PARAM_MESSAGE_2_CARRY_2, size);
+    ///
+    /// let msg = 230;
+    /// let scalar = 376;
+    ///
+    /// let mut ct = cks.encrypt(msg);
+    ///
+    /// // Compute homomorphically a scalar multiplication:
+    /// let ct_res = sks.scalar_mul_parallelized(&mut ct, scalar);
+    ///
+    /// // Decrypt:
+    /// let clear: u64 = cks.decrypt(&ct_res);
+    /// assert_eq!(msg * scalar % modulus, clear);
+    /// ```
+    pub fn scalar_mul_parallelized<PBSOrder: PBSOrderMarker>(
+        &self,
+        ct: &RadixCiphertext<PBSOrder>,
+        scalar: u64,
+    ) -> RadixCiphertext<PBSOrder> {
+        let mut ct_res = ct.clone();
+        self.scalar_mul_assign_parallelized(&mut ct_res, scalar);
+        ct_res
+    }
+
+    pub fn scalar_mul_assign_parallelized<PBSOrder: PBSOrderMarker>(
+        &self,
+        ct: &mut RadixCiphertext<PBSOrder>,
+        scalar: u64,
+    ) {
+        let zero = self.create_trivial_zero_radix(ct.blocks.len());
```

Comment:
its the same code as smart_scalar_mul_assign ? could we call `smart_scalar_mul_parallelized` is has a compatible signature

Reply:
not strictly the reduction is done via the default add to avoid variable performance 😭

Reply:
ah i missed that